### PR TITLE
fix: Support build log URLs on GitLab and BitBucket Server

### LINKED ISSUE #6834: jx tries to retrieve the logs from raw.githubusercontent no matter the provider
### Summary 
jx tries to retrieve the logs from raw.githubusercontent no matter the git provider

### Steps to reproduce the behaviour
clone a project from gitlab or bitbucket
```
jx import
jx get build log
```

### Expected behavior
logs should be retieved from https://gitlab.com/<org>/<project>/-/raw/gh-pages/jenkins-x/logs/

### Actual behavior
jx fails to retrieved logs since it 's checking in the wrong location

### Jx version

The output of `jx version` is:

```
jx                 2.0.1202
Kubernetes cluster v1.15.2
kubectl            v1.17.0
git                2.21.1
Operating System   "Fedora release 30 (Thirty)"
```

### Jenkins type

- [X] Serverless Jenkins X Pipelines (Tekton + Prow)
- [ ] Classic Jenkins

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #6834

### CODE REVIEW

File: pkg/collector/git_collector.go [hunk 1]
```diff
@@ -162,8 +165,15 @@ func (c *GitCollector) generateURL(storageOrg string, storageRepoName string, rP
...
-		// TODO only supporting github for now!!!
-		url = fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/%s/%s", storageOrg, storageRepoName, c.gitBranch, rPath)
+		switch c.gitKind {
+		case gits.KindGitlab:
+			url = fmt.Sprintf("https://%s/%s/%s/-/raw/%s/%s", c.gitInfo.Host, storageOrg, storageRepoName, c.gitBranch, rPath)
+		case gits.KindBitBucketServer:
+			url = fmt.Sprintf("https://%s/projects/%s/repos/%s/raw/%s?at=%s", c.gitInfo.Host, storageOrg, storageRepoName, rPath, neturl.QueryEscape(fmt.Sprintf("refs/heads/%s", c.gitBranch)))
+		default:
+			// Fall back on GitHub for now
+			url = fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/%s/%s", storageOrg, storageRepoName, c.gitBranch, rPath)
```
Comment (mrageh) [inline pkg/collector/git_collector.go:175]: What happens if the `gitKind` is not `GitHub`? 

Would this code not break?

Comment (abayer) [reply pkg/collector/git_collector.go:175]: Well, yeah, but it's already broken. =) We handle the GitHub Enterprise case in the first part of this, before the `switch` I've added, and until now, the only alternative was this line. With the `switch`, we at least check GitLab and BitBucket Server cases and construct the appropriate URLs for those.

**Pull request merged**